Feature/grades related pages

DavidWilliamK opened the pull request:
Automate tests for scoringLandingPage

DavidWilliamK committed:
Added automation for grades landing page

DavidWilliamK committed:
Refactor scoring related tests

DavidWilliamK committed:
Updated annotation name

Change @QuestionBankList to @GradesItemList

DavidWilliamK committed:
Resolve Conflicts

oliverrsebastian commented on src/main/java/com/future/function/qa/pages/base/MenuPage.java:
```diff
@@ -15,6 +15,11 @@ public WebElementFacade comparisonsMenu(int menuIndex) {
 
+  public WebElementFacade gradesMenu(int menuIndex) {
+
+    return find(By.xpath(String.format("//body//li[%s]", menuIndex)));
```
you can use karnando's code in HomePage where you can get the menu with its index

DavidWilliamK replied:
Done

oliverrsebastian commented on src/test/resources/features/scoring.Landing/Grades.feature:
```diff
@@ -0,0 +1,167 @@
+    Then user should be in grades page
+    Then user should see grades tab list item with index 0 in grades menu
+    And user should see add button on tab 0
+    And user should see action dropdown button on tab 0 item with index 0
```
maybe assert if tab 0 has class `is-active`

DavidWilliamK replied:
Assertion is added to step `Then user should see grades tab list item with index 0 in grades menu`
Is this fine? @oliverrsebastian

oliverrsebastian replied:
i think it's better to have it as a step in this feature file because we can define the class we wanted to assert instead of hardcoded it in Steps file
cc @DavidWilliamK

DavidWilliamK committed:
Added new assertion and removed redundant function

Now use the pre-defined function to access pages from menuList by index
Added is-active class assertion to each tab with specified index

DavidWilliamK committed:
Added new assertion step

Added is-active class assertion to each tab with specified index
